platform/centec/libsaithrift-dev.mk

```diff
@@ -4,15 +4,16 @@ SAI_VER = 0.9.4
 
 LIBSAITHRIFT_DEV = libsaithrift-dev_$(SAI_VER)_amd64.deb
 $(LIBSAITHRIFT_DEV)_SRC_PATH = $(SRC_PATH)/sonic-sairedis/SAI
-$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT) $(LIBTHRIFT_DEV) $(PYTHON_THRIFT) $(THRIFT_COMPILER) $(CENTEC_SAI)
+$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT) $(LIBTHRIFT_DEV) $(PYTHON_THRIFT) $(THRIFT_COMPILER) $(CENTEC_SAI) $(LIBHIREDIS) $(LIBNL3_DEV) $(LIBNL_GENL3_DEV)
```

Comment:
just want to make sure, the saiserver really needs this 'LIBHIREDIS' liberary?
The saiserver just tries to expose the sai api as PRC and should be run in an environment without any other databases from upper layer.

this libsaithrift-dev.mk will be referenced by other platforms on centec, and the depends might be installed on other platforms, just want to make sure, if those liberaries, $(LIBNL3_DEV) $(LIBNL_GENL3_DEV), also needed by other centec platforms, and you verified on them? 
Or you can just added those liberaries to the related platforms in the corresponding docker-sai*.mk or docker-sync*.mk

Reply:
Thanks  for the feedback.  
Centec SAI needs "LIBHIREDIS", "LIBNL3_DEV" and "LIBNL_GENL3_DEV". LIBHIREDIS is used for warmboot which will store and recover SAI software data. Othre vendor may not use redis to implement SAI warmboot. "LIBNL3_DEV" and "LIBNL_GENL3_DEV" is used for sFlow.

Saiserver will link SAI library which is depend on "LIBHIREDIS", "LIBNL3_DEV" and "LIBNL_GENL3_DEV". 

I think that other centec platforms should also need these libraries according to the centec SAI implement. If we remove these libraries from libsaithrift-dev.mk, syncd-rpc may compile link fail. 
The fail log is as following:
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `genl_ctrl_resolve_grp@libnl_3'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_connect'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_socket_alloc'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisConnect'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisAppendCommand'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_socket_free'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisCommand'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisGetReply'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `freeReplyObject'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisFree'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `redisCommandArgv'
/usr/bin/ld: /usr/lib/libsai.so: undefined reference to `nl_socket_disable_seq_check'
collect2: error: ld returned 1 exit status